client/src/components/Map/Map.jsx

```diff
@@ -0,0 +1,151 @@
+// Imports
+import React, { Component } from 'react';
+
+// Import React Script Library to load Google object
+import Script from 'react-load-script';
+
+import MapCards from './MapCards';
+
+class Map extends Component {
+  constructor(props) {
+    super(props);
+
+    // Placeholder state
+    this.initialState = {
+      locations: [
+        {
+          name: '',
+          place_Id: '' 
+        }
+      ] 
+    };
+
+    this.state = this.initialState
+    
+    // Bind Functions
+    this.handleScriptLoad = this.handleScriptLoad.bind(this);
+    this.handleMapChange = this.handleMapChange.bind(this);
+  }
+
+
+  handleScriptLoad() {
+    // Try HTML5 Geolocation
+    if (navigator.geolocation) {
+    
+      navigator.geolocation.getCurrentPosition(
+        position => {
+          let pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+          };
+          // Loads map
+          let map = new google.maps.Map(document.getElementById('map'), {
+            center: pos,
+            zoom: 13
+          });
+        }, () => {
+          // If user denies geolocation info, default location is used
+          this.handleLocationError(false)
+        }
+      )
+    }
+
+    // Initialize Google Autocomplete
+    /*global google*/ // To disable any eslint 'google not defined' errors
+    this.autocomplete = new google.maps.places.Autocomplete(document.getElementById('autocomplete'));
+
+    // Sets autocomplete fields to be returned
+    this.autocomplete.setFields(['address_components', 'formatted_address', 'geometry', 'icon', 'name', 'place_id']);
+    
+    // When a new place is selected the map will be forced to update
+    this.autocomplete.addListener('place_changed', this.handleMapChange);
+  }    
+
+  handleLocationError(browserHasGeolocation) {
+    // Set default location to Sydney, Australia
+    let pos = { lat: -33.856, lng: 151.215 };
+
+    let map = new google.maps.Map(document.getElementById("map"), {
+      center: pos,
+      zoom: 15
+    });
+  }
+
```

Comment:
Nice job. A walk through would be helpful if not yet done, as far the maps api. A question I have would be if there are other fields that can be returned for the auto complete that would be useful to us.  But the current set fields are good basics for release 1.

Reply:
We can have it return an address, reviews, rating, photos, phone number, website (if they have one), hours, price level. We could also make a drop-down to search by business type.  I could also change the query type to be dynamic so that a user can search for whatever type of venue they want. For example, they could just search for 'Starbucks' instead of having it give them nearby cafes.